etcd-aws-cluster

```diff
@@ -34,14 +38,18 @@ fi
 # If we're in proxy mode we don't have to look this up and expect an env var
 if [[ ! $PROXY_ASG ]]; then
     etcd_proxy=off
-    asg_name=$(aws autoscaling describe-auto-scaling-groups --region $region | jq --raw-output ".[] | map(select(.Instances[].InstanceId | contains(\"$ec2_instance_id\"))) | .[].AutoScalingGroupName")
+    asg_name=$(aws autoscaling describe-auto-scaling-groups | jq --raw-output ".[] | map(select(.Instances[].InstanceId | contains(\"$ec2_instance_id\"))) | .[].AutoScalingGroupName")
     if [[ ! $asg_name ]]; then
         echo "$pkg: failed to get the auto scaling group name"
         exit 4
     fi
 else
     etcd_proxy=on
-    asg_name=$PROXY_ASG
+    if [[ -n $ASG_BY_TAG ]]; then
+        asg_name=$(aws autoscaling describe-auto-scaling-groups | jq --raw-output ".[] | map(select(.Tags[].Value == \"$PROXY_ASG\")) | .[].AutoScalingGroupName")
```

Comment:
I'm wondering how this might work in the case where there is more than 1 ASG with the same tag? Using tags feels a little brittle.